llvm/lib/Target/X86/Disassembler/X86Disassembler.cpp

```diff
@@ -1219,6 +1219,9 @@ static int getInstructionID(struct InternalInstruction *insn,
         attrMask |= ATTR_EVEXKZ;
       if (bFromEVEX4of4(insn->vectorExtensionPrefix[3]))
         attrMask |= ATTR_EVEXB;
+      if (x2FromEVEX3of4(insn->vectorExtensionPrefix[2]) &&
+          (insn->opcodeType != MAP4))
```

Comment:
I think you also need to check more things for this, e.g. `EVEX.b`. Since the APX SPEC says

> When any of the bits in EVEX.{R4,X4,B4} is not used by a promoted VEX instruction, it is ignored, 

@FreddyLeaf is working on adding tests for the ignore semantic while it's not ready yet.

Reply:
Thanks for the reminder. It's fully matched with AVX10.2 SPEC now.